apps/site/components/withDownloadArchive.tsx

```diff
@@ -0,0 +1,47 @@
+import type { FC } from 'react';
+
+import { getClientContext } from '#site/client-context';
+import getReleaseData from '#site/next-data/releaseData';
+import {
+  buildReleaseArtifacts,
+  extractVersionFromPath,
+  findReleaseByVersion,
+} from '#site/util/downloadUtils/archive';
+
+type DownloadArchive = ReturnType<typeof buildReleaseArtifacts>;
+
+type WithDownloadArchiveProps = {
+  children: FC<DownloadArchive & { majorVersions: Array<string> }>;
+};
+
+/**
+ * Higher-order component that extracts version from pathname,
+ * fetches release data, and provides download artifacts to child component
+ */
+const WithDownloadArchive: FC<WithDownloadArchiveProps> = async ({
+  children: Component,
+}) => {
+  const { pathname } = getClientContext();
+  const releaseData = await getReleaseData();
```

Comment:
Can we move this statement to after the version check? That way, we don't call `getReleaseData` when `version == null` (since it'll be unused).